rust/src/measurements/gaussian/float/mod.rs

```diff
@@ -95,11 +105,20 @@ where
         relaxation = relaxation.inf_mul(&T::inf_cast(size)?)?;
     }
 
+    let relaxation = f64::inf_cast(relaxation)?;
+    let r_scale = scale.into_rational()?;
+
     Measurement::new(
         input_domain,
         Function::new_fallible(move |arg: &Vec<T>| {
             arg.iter()
-                .map(|shift| sample_discrete_gaussian_Z2k(shift.clone(), scale, k))
+                .map(|shift| {
+                    let sample =
+                        sample_discrete_gaussian_Z2k(shift.into_rational()?, r_scale.clone(), k)?;
+
+                    // postprocessing: round to nearest T
+                    Ok(T::from_rational(sample))
```

Comment:
Here and on line 57 it adds a `from_rational`: Not really sure why it's needed now and not before.

Reply:
I changed `sample_discrete_xxx_Z2k` to just be rational in -> rational out. It used to also do conversions to/from floats.